### PR TITLE
ansible-connector: add group vars to each host

### PR DESCRIPTION
**Pull request by blinkinglight**
ansible hosts.yaml could have vars in each group. so add these group vars to each host. sorry, i am not familar with python but maybe this would be good start to someone.

//TODO: implement global varaibles for all hosts and override if key match

example of confing:
```
all:
  vars:
    hostname: default.hostname.to.all
prod:
  hosts:
    my.host.name.tld:
  vars:
    hostname: custom.for.hosts
    vhosts:
        vhost1:
             host: ip1.addr.1.1.1.1
```
etc.

### CODE REVIEW

File: pyinfra/api/connectors/ansible.py [hunk 1]
```diff
@@ -133,23 +143,46 @@ def _parse_ansible_hosts(hosts):
...
             # recursively parse inventory tree
-            parse_inventory_tree(inventory_tree[group]['children'], host_to_groups, groups)
+            parse_inventory_tree(inventory_tree[group]['children'], host_to_groups, groups, group_varz)
 
     return host_to_groups
 
 
-def append_groups_to_host(host, groups, host_to_groups):
+
+def parse_inventory_tree_vars(inventory_tree, host_to_groups=dict(), group_stack=set(), group_varz=dict()):
```
Comment (Fizzadar) [inline pyinfra/api/connectors/ansible.py:166]: I'm not 100% sure this function is needed? I think the `parse_inventory_tree` could also handle this (returning group vars as well as host to groups).

File: pyinfra/api/connectors/ansible.py [hunk 2]
```diff
@@ -133,23 +143,46 @@ def _parse_ansible_hosts(hosts):
             yield host
 
 
-def parse_inventory_tree(inventory_tree, host_to_groups=dict(), group_stack=set()):
+def parse_inventory_tree(inventory_tree, host_to_groups=dict(), group_stack=set(), group_varz=dict()):
```
Comment (Fizzadar) [inline pyinfra/api/connectors/ansible.py:146]: Not spotted these before - these default mutable arguments need to be replaced (not that they are introduced by this PR)!

**Review by Fizzadar — COMMENTED** [review]
@blinkinglight apologies for the delay in getting back to you on this, keen to get support for this added to `pyinfra`. Happy to have a look myself if you can add me as a collaborator on your fork?

**Comment by blinkinglight** [comment]
> @blinkinglight apologies for the delay in getting back to you on this, keen to get support for this added to `pyinfra`. Happy to have a look myself if you can add me as a collaborator on your fork?

hi, invite sent.

**Comment by Fizzadar** [comment]
Cancelling this out, not a priority and the Ansible connector will be dropped in v3 (if desired others can implement a connector as a separate module).

**Pull request closed without merging**